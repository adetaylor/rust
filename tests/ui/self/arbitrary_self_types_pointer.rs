// run-pass

#![feature(receiver_trait)]

struct A;

impl A {
    fn m(self: *const Self) {}
}

trait B {
    fn bm(self: *const Self) {}
}

fn main() {
    let a = A;
    let ptr = &a as *const A;
    ptr.m();
}
