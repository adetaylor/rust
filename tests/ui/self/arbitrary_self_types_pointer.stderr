error[E0307]: invalid `self` parameter type: *const A
  --> $DIR/arbitrary_self_types_pointer.rs:6:16
   |
LL |     fn m(self: *const Self) {}
   |                ^^^^^^^^^^^
   |
   = note: type of `self` must be `Self` or some type implementing Receiver
   = help: consider changing to `self`, `&self`, `&mut self`, or a type implementing `Receiver`: for example `self: Box<Self>`, `self: Rc<Self>`, `self: Arc<Self>`, or `self: Pin<P>` (where P is one of the previous types except `Self`). Raw pointers do not implement Receiver. Consider wrapping your pointer in a newtype wrapper for which you implement Receiver.

error[E0307]: invalid `self` parameter type: *const Self
  --> $DIR/arbitrary_self_types_pointer.rs:11:17
   |
LL |     fn bm(self: *const Self) {}
   |                 ^^^^^^^^^^^
   |
   = note: type of `self` must be `Self` or some type implementing Receiver
   = help: consider changing to `self`, `&self`, `&mut self`, or a type implementing `Receiver`: for example `self: Box<Self>`, `self: Rc<Self>`, `self: Arc<Self>`, or `self: Pin<P>` (where P is one of the previous types except `Self`). Raw pointers do not implement Receiver. Consider wrapping your pointer in a newtype wrapper for which you implement Receiver.

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0307`.
